tsec-http4s/src/main/scala/tsec/authentication/SignedCookieAuthenticator.scala

```diff
@@ -15,10 +15,118 @@ import tsec.mac.jca._
 
 import scala.concurrent.duration.FiniteDuration
 
-abstract class SignedCookieAuthenticator[F[_]: Sync, I, V, Alg] private[tsec] (
-    val expiry: FiniteDuration,
-    val maxIdle: Option[FiniteDuration]
-) extends AuthenticatorService[F, I, V, AuthenticatedCookie[Alg, I]]
+abstract class SignedCookieAuthenticator[F[_], I, V, Alg] private[tsec] (
+    val tokenStore: BackingStore[F, UUID, AuthenticatedCookie[Alg, I]],
+    val idStore: IdentityStore[F, I, V],
+    val settings: TSecCookieSettings
+)(implicit F: Sync[F])
+    extends Authenticator[F, I, V, AuthenticatedCookie[Alg, I]] {
+
+  val expiry: FiniteDuration = settings.expiryDuration
+
+  val maxIdle: Option[FiniteDuration] = settings.maxIdle
+
+  /** Generate a nonce by concatenating the message to be sent with the current instant and hashing their result
+    * Possibly this should be a variable argument, but for now SHA1 is enough, since the chance for collision is
+    * abysmally low.
+    */
+  private def generateNonce(message: String) =
```

Comment:
should return `F[]`

Reply:
Actually no. Hashing on the JCA at least is pure. This doesn't mutate anything

Reply:
I take that back. You are absolutely right about this. I didn't see the Instant.now call, i wrote that a while ago. My bad. Will fix